pkg/asset/machines/aws/master.go

```diff
@@ -2,72 +2,106 @@
 package aws
 
 import (
-	"text/template"
+	"fmt"
+
+	"github.com/openshift/installer/pkg/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	clusterapi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-// MasterConfig is used to generate master machines
-type MasterConfig struct {
-	MachineConfig
-	Instances []MasterInstance
-}
+// MasterMachines returns a list of master machines for a machinepool.
+func MasterMachines(config *types.InstallConfig, pool *types.MachinePool, userDataSecret string) ([]clusterapi.Machine, error) {
+	var out []clusterapi.Machine
+	if pool.Name != "master" {
+		return nil, fmt.Errorf("only pool name master is supported")
```

Comment:
Do we care?  I'm fine allowing folks to call their master pool something else, although we may want [this][1] so *we* can handle it.  Still, that would happen above the level of this helper, so I don't see a need to guard here.

[1]: https://github.com/openshift/installer/issues/264#issuecomment-422349518

Reply:
Yes I liked that this PR allows us to still do that change to separate control plane and compute clearly, whichout changing these functions.

Reply:
> Yes I liked that this PR allows us to still do that change...

I'm having trouble parsing this.  @dgoodwin, are you saying "yes, I like the `Name != master` check" or "yes, I like the idea of dropping that check"?